### ISSUE TITLE
prefer-screen-queries: Don't warn if using the baseElement option

### ISSUE THREAD

**Issue by kentcdodds** [testing-library/eslint-plugin-testing-library#188]
This should be acceptable (from the React Testing Library tests):

```javascript
test('baseElement isolates trees from one another', () => {
  const {getByText: getByTextInA} = render(<div>Jekyll</div>, {
    baseElement: treeA,
  })
  const {getByText: getByTextInB} = render(<div>Hyde</div>, {
    baseElement: treeB,
  })

  expect(() => getByTextInA('Jekyll')).not.toThrow(
    'Unable to find an element with the text: Jekyll.',
  )
  expect(() => getByTextInB('Jekyll')).toThrow(
    'Unable to find an element with the text: Jekyll.',
  )

  expect(() => getByTextInA('Hyde')).toThrow(
    'Unable to find an element with the text: Hyde.',
  )
  expect(() => getByTextInB('Hyde')).not.toThrow(
    'Unable to find an element with the text: Hyde.',
  )
})
```

**Comment by Belco90** [comment]
So the use case is basically several renders within same test right? Just trying to find a pattern to apply to the rule.

**Comment by timdeschryver** [comment]
I think we should take `baseElement` into account in some of the rules and disable them? (mostly the rules that are working with `screen`).

**Comment by thomlom** [comment]
If I get it right, we need to make it so `prefer-screen-queries` doesn't fire if the `render` function has set the `container` or `baseElement` option. Is that correct?

**Comment by kentcdodds** [comment]
That is correct

**Comment by gndelia** [comment]
I **think** I can take this one if no one else is taking it

**Comment by kentcdodds** [comment]
Sounds awesome 🤘
